test/e2e/flowaggregator_test.go

```diff
@@ -102,18 +102,33 @@ const (
 )
 
 func TestFlowAggregator(t *testing.T) {
-	skipIfDualStackCluster(t)
-	data, isIPv6, err := setupTestWithIPFIXCollector(t)
+	data, v4Enabled, v6Enabled, err := setupTestWithIPFIXCollector(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
 	defer teardownFlowAggregator(t, data)
 
-	podAIP, podBIP, podCIP, svcB, svcC, err := createPerftestPods(data, isIPv6)
+	podAIP, podBIP, podCIP, err := createPerftestPods(data)
 	if err != nil {
-		t.Fatalf("Error when creating perftest pods and services: %v", err)
+		t.Fatalf("Error when creating perftest pods: %v", err)
 	}
+
+	if v4Enabled {
+		createSvcRunTests(t, data, podAIP, podBIP, podCIP, false)
+	}
+
+	if v6Enabled {
+		createSvcRunTests(t, data, podAIP, podBIP, podCIP, true)
+	}
+}
+
+func createSvcRunTests(t *testing.T, data *TestData, podAIP *PodIPs, podBIP *PodIPs, podCIP *PodIPs, isIPv6 bool) {
```

Comment:
Separate out createServices function and runTests function for better readability

Reply:
Thanks, addressed.